crates/bevy_ecs/src/system/exclusive_function_system.rs

```diff
@@ -121,6 +121,7 @@ where
         self.system_meta.last_run.set(*change_tick);
         *change_tick = change_tick.wrapping_add(1);
         world.last_change_tick = saved_last_tick;
+        world.flush_commands();
```

Comment:
We probably want to call `flush_commands` *before* we revert the change tick right? That way any hooks will be called within the tick of the exclusive system that triggered them.